core/txflow/src/dag/mod.rs

```diff
@@ -1,19 +1,21 @@
 mod message;
+mod reporter;
 
 use primitives::traits::{Payload, WitnessSelector};
 use primitives::types::*;
 
 use std::collections::HashSet;
 
 use self::message::Message;
+pub use self::reporter::*;
 use typed_arena::Arena;
 
 /// The data-structure of the TxFlow DAG that supports adding messages and updating counters/flags,
 /// but does not support communication-related logic. Also does verification of the messages
-/// received from other nodes.
+/// received from other nodes and store detected violations.
 /// It uses unsafe code to implement a self-referential struct and the interface makes sure that
 /// the references never outlive the instances.
-pub struct DAG<'a, P: 'a + Payload, W: 'a + WitnessSelector> {
+pub struct DAG<'a, P: 'a + Payload, W: 'a + WitnessSelector, M: 'a + MisbehaviourReporter> {
```

Comment:
Please, create a NoopMisbehaviorReporter that does nothing, then set it as a default type param so that our existing tests do not need modifications:
DAG<'a, P: 'a + Payload, W: 'a + WitnessSelector, M: 'a + MisbehaviourReporter = NoopMisbehaviorReporter>

Reply:
Ok, I'll remove FakeMisbehaviourReporter, as I can use Noop*. Maybe at some point we might need Fake (for any extra func).